### ISSUE TITLE
Can we add pattern similar to EndpointBaseSync.WithRequest<T>.WithResponse<U> (ardalis)

### ISSUE THREAD

**Issue by ghuserast** [FastEndpoints/FastEndpoints#779]
Hi @ardalis ,

I really loved the pattern implemented here [EndpointBaseAsync.cs](https://github.com/ardalis/ApiEndpoints/blob/main/src/Ardalis.ApiEndpoints/FluentGenerics/EndpointBaseAsync.cs) by @ardalis.

It would be great to have that pattern implemented for FastEndpoints as well.

Thanks!

**Comment by ardalis** [comment]
You just want the `.` notation as apposed to `Endpoint<TRequest,TResponse>` and `EndpointWithoutRequest<TResponse>` etc?

**Comment by dj-nitehawk** [comment]
implemented in `v5.29.0.13-beta`

```csharp
sealed class MyEndpoint : Ep.Req<MyRequest>.Res<MyResponse>.Map<MyMapper>
{
    ...
}

sealed class MyEndpoint : Ep.Req<MyRequest>.NoRes.Map<MyMapper>
{
    ...
}

sealed class MyEndpoint : Ep.NoReq.Res<MyResponse>
{
    ...
}
```
